OKlib.js

```diff
@@ -824,416 +767,378 @@ registerPlugin({
+     * returns the elements that differ are contained in only one of the given arrays
+     * @param {Object[]} array
+     * @param {Object[]} elements
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Object[]} an array that contains the elements that were only present in one of the arrays
      */
     function arrayDifference(array, elements, compare) {
         let result = [];
-        for (let i = 0; i < elements.length; i++) {
-            if (!arrayContainsElement(array, elements[i], compare)) {
-                result.push(elements[i]);
-            }
+        for (let i in elements) {
+            if (!arrayContainsElement(array, elements[i], compare)) result.push(elements[i]);
         }
-        for (let j = 0; j < array.length; j++) {
-            if (!arrayContainsElement(elements, array[j], compare)) {
-                result.push(array[j]);
-            }
+        for (let j in array) {
+            if (!arrayContainsElement(elements, array[j], compare)) result.push(array[j]);
         }
         return result;
     }
 
     /**
-     * Gets the Index of an Object in an Array or -1 if it is not contained
-     * @param  {Object[]} array
-     * @param  {Object} element
-     * @param  {Function} compare  A Compare Function that should be used for the Comparison, if not set 'equal' is used
-     * @return {Integer}         [description]
+     * gets the index of an object in an array or -1 if it is not contained
+     * @param {Object[]} array
+     * @param {Object} element
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Integer} [description]
      */
     function arrayGetIndex(array, element, compare) {
-        if (!compare) {
-            compare = equal;
-        }
-        for (let i = 0; i < array.length; i++) {
-            if (compare(array[i], element)) {
-                return i;
-            }
+        if (!compare) compare = equal;
+        for (let i in array) {
+            if (compare(array[i], element)) return i;
         }
         return -1;
     }
 
     /**
-     * Removes undefined and null entries from an Array
-     * @param  {Object[]} array
-     * @return {Object[]} A Copy of the Array with any undefined or null entries removed
+     * removes undefined and null entries from an array
+     * @param {Object[]} array
+     * @return {Object[]} a copy of the array with any undefined or null entries removed
      */
     function arrayRemoveUndefined(array) {
         array = arrayCreateArray(array);
         let result = [];
-        array.forEach(element => {
-            if (!empty(element)) {
-                result.push(element);
-            }
-        })
-        log("arrayRemoveUndefined: Removed '" + (array.length - result.length) + "' undefined or null entries from the array", 4);
+        for (let i in array) {
```

Comment:
change this to for..of or revert to normal for loop, because for..in is not safe for array operation (might not iterate in the right order)